### PR TITLE
#1493 refactoring remove any type from srcfrontendsrcpagesfinancepagereimbursementrequestformreimbursementproducttabletsx

### LINKED ISSUE #1493: [Refactoring] - Remove `any` type from `src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementProductTable.tsx`
### Description

Remove the use of the `any` type from ReimbursementProductTable.tsx This may seem very simple, but the point of this is just to have you read through the codebase and understand the types/interfaces used in it, or make new ones that fit the shape of the data being used.

### Notes

Check #665 for notes on why we're doing this and pointers on what you should replace `any` with.

### Acceptance Criteria

`any` is not used as a type in ReimbursementProductTable.tsx

### PR DESCRIPTION
**Pull request by JadenZhub**
## Changes

Changed any to ReimbursementProductFormArgs

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1493

### CODE REVIEW

**Review by martin0he — APPROVED** [review]
lgtm

**Review by martin0he — CHANGES_REQUESTED** [review]
when you open a pr, make sure you tick off all the boxes in the checklist before asking for a review.

File: src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementProductTable.tsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ interface ReimbursementProductTableProps {
     id: string;
   }[];
   errors: FieldErrors<ReimbursementRequestFormInput>;
-  control: Control<ReimbursementRequestFormInput, any>;
+  control: Control<ReimbursementRequestFormInput, ReimbursementProductFormArgs>;
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementProductTable.tsx:47]: you can delete the `, any` and it should work still work. Let's do it that way cause it's a bit simpler

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementProductTable.tsx (hunk 1 above)
Comment (JadenZhub) [reply src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementProductTable.tsx]: done ✅

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**